internal/util/debug/debug.go

```diff
@@ -56,11 +62,59 @@ func RunHandler(ctx context.Context, addr string, r prometheus.Registerer, l *za
 	}
 	must.NoError(statsviz.Register(http.DefaultServeMux, opts...))
 
+	requestCount := prometheus.NewCounterVec(
```

Comment:
Have you tested this?

I don't see where we increment this counter. Also, `collect` and `decribe` seem missing. 

And also, I think @AlekSi meant we could use this https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promhttp library.